src/DSE.Open/AsciiChar2.cs

```diff
@@ -264,12 +350,11 @@ public static AsciiChar2 Parse(ReadOnlySpan<byte> utf8Text, IFormatProvider? pro
     public static bool TryParse(
         ReadOnlySpan<byte> utf8Text,
         IFormatProvider? provider,
-        [MaybeNullWhen(false)] out AsciiChar2 result
-    )
+        out AsciiChar2 result)
     {
-        if (utf8Text.Length == MaxSerializedByteLength)
+        if (utf8Text.Length == MaxSerializedByteLength && AsciiChar.IsAscii(utf8Text[0]) && AsciiChar.IsAscii(utf8Text[1]))
         {
-            result = new AsciiChar2((AsciiChar)utf8Text[0], (AsciiChar)utf8Text[1]);
```

Comment:
This would have thrown if passed `char >= 128`. Added a test but could do with more coverage of this and AsciiChar3